### PR TITLE
Change generic-type algorithm

### LINKED ISSUE #493: Generic upper bound specification warns about [GENERIC_NAME] lint.
## Steps to Reproduce
1. Paste the following code:
```kotlin
interface Test<T : Collection<T>>
```
2. Run the diktat with default configurations.

## Expected behavior
There shouldn't be any warnings.

## Observed behavior
The following warning is encountered:
```kotlin
[GENERIC_NAME] generic name should contain only one single capital letter, it can be followed by a number: <T : Collection<T>>
```

## Environment information
* diktat version: **0.1.2**
* build tool (maven/gradle): **gradle**
* how is diktat run (CLI, plugin, etc.): **gradle task**
* kotlin version: **1.4.10**
* operating system: **Windows 10 Build 1909**

### PR DESCRIPTION
**Pull request by kentr0w**
### What's done:
    Changed algorithm
   closes #493

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt [hunk 1]
```diff
@@ -362,13 +343,12 @@ class IdentifierNaming(private val configRules: List<RulesConfig>) : Rule("ident
...
-        // removing whitespaces and <>
-        val genericName = generic.replace(">", "").replace("<", "").trim()
-        // first letter should always be a capital
-        return genericName[0] in 'A'..'Z' &&
-                // other letters - are digits
-                (genericName.length == 1 || genericName.substring(1).isDigits())
+    private fun validGenericTypeName(generic: ASTNode): Boolean {
+        generic.findChildByType(TYPE_PARAMETER)?.let {
+            val typeText = it.getIdentifierName()?.text ?: return false
+            return typeText[0] in 'A'..'Z' &&  (typeText.length == 1 || typeText.substring(1).isDigits())
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt:349]: Please keep comments to these conditions

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt [hunk 2]
```diff
@@ -362,13 +343,12 @@ class IdentifierNaming(private val configRules: List<RulesConfig>) : Rule("ident
...
      */
-    private fun validGenericTypeName(generic: String): Boolean {
-        // removing whitespaces and <>
-        val genericName = generic.replace(">", "").replace("<", "").trim()
-        // first letter should always be a capital
-        return genericName[0] in 'A'..'Z' &&
-                // other letters - are digits
-                (genericName.length == 1 || genericName.substring(1).isDigits())
+    private fun validGenericTypeName(generic: ASTNode): Boolean {
+        generic.findChildByType(TYPE_PARAMETER)?.let {
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt:347]: There should be several type parameters, we should check all of them

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt [hunk 3]
```diff
@@ -28,27 +29,7 @@ import org.cqfn.diktat.ruleset.constants.Warnings.OBJECT_NAME_INCORRECT
...
-import org.cqfn.diktat.ruleset.utils.isDigits
-import org.cqfn.diktat.ruleset.utils.isLowerCamelCase
-import org.cqfn.diktat.ruleset.utils.isPascalCase
-import org.cqfn.diktat.ruleset.utils.isUpperSnakeCase
-import org.cqfn.diktat.ruleset.utils.removePrefix
-import org.cqfn.diktat.ruleset.utils.toLowerCamelCase
-import org.cqfn.diktat.ruleset.utils.toPascalCase
-import org.cqfn.diktat.ruleset.utils.toUpperSnakeCase
-import org.cqfn.diktat.ruleset.utils.Style
+import org.cqfn.diktat.ruleset.utils.*
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt:32]: Please un-collapse imports :)

**Review by petertrr — CHANGES_REQUESTED** [review]

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter1/IdentifierNamingWarnTest.kt [hunk 4]
```diff
@@ -592,4 +592,23 @@ class IdentifierNamingWarnTest : LintTestBase(::IdentifierNaming) {
...
+    @Test
+    @Tag(WarningNames.GENERIC_NAME)
+    fun `check generic types`() {
+        val code =
+                """
+                    interface Test<String>
+                    interface Test1<T: String>
+                    interface Test2<T : Collection<T>>
+                    interface Test3<out T>
+                    interface Test3<in T>
```
Comment (petertrr) [inline diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter1/IdentifierNamingWarnTest.kt:605]: What about `in Tr`, will it be checked correctly?

**Review by petertrr — CHANGES_REQUESTED** [review]

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter1/IdentifierNamingWarnTest.kt (hunk 4 above)
Comment (kentr0w) [reply diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter1/IdentifierNamingWarnTest.kt:605]: It'll trigger. I added tests.

**Review by petertrr — APPROVED** [review]
lgtm

**Pull request merged**